airbyte-commons-server/src/main/java/io/airbyte/commons/server/handlers/OAuthHandler.java

```diff
@@ -361,4 +362,27 @@ JsonNode getOAuthInputConfiguration(final JsonNode hydratedSourceConnectionConfi
     return Jsons.jsonNode(result);
   }
 
+  public Map<String, Object> writeOAuthSecret(final UUID workspaceId, final Map<String, Object> payload) {
+
+    try {
+      final String payloadString = Jackson.getObjectMapper().writeValueAsString(payload);
+      final SecretCoordinate secretCoordinate = secretsRepositoryWriter.storeSecret(generateOAuthSecretCoordinate(workspaceId), payloadString);
+      return formatOAuthSecretResponse(secretCoordinate);
```

Comment:
Two questions:

1. It seems like we're writing the full OAuth response, stringified, in one coordinate. On the usual creation flow, each secret gets its own coordinate. Should we follow the same approach to make things easier when it comes to creating the source with these credentials? Note that some sources return multiple secret (and non-secret) values as part of the complete_oauth response. A good example is any source with single-use refresh tokens, like GitLab, which returns access_token (secret), refresh_token (secret) and a token_expiry_date (not secret)
2. Do we want to strip out any masked instance-wide params? (I believe they would be coming through here as asterisks (`******`) )

Reply:
Might make sense to have a quick chat with you about this since I think it might be pretty illuminating for me, but generically I think part of the answer to both of your points above is to potentially port some of the frontend code to backend.

It seems like frontend has the connector specification handy and when receiving the OAuth response, uses the specification to determine what properties from the response it needs to hold onto/use in source creation.

My inclination is to snag the spec on the backend, follow the same logic that the frontend follows and save the necessary OAuth params as a single secret.

When we create the source, we would fetch the OAuth params via the secret coordinate and then re-save in the way you describe. In our discussions of this Evan (I think?) thought that it was perfectly fine for us to create new secrets as part of source/destination creation and not use the secrets created by this process to directly create sources/destinations.

Does the above seem reasonable?

Reply:
Chatted w/ @pedroslopez about this.  Pedro - please add anything I missed!

Where we landed was:
1. As soon as reasonably possible, we should consider adding a check operation to this process to ensure people can't create/update sources/destinations with configurations that validate but can't actually be used to connect to the source/destination. As such, we might benefit from bundling check into source create/update at least externally.
    - This has the added benefit of potentially allowing us to handle the case where single use refresh tokens are consumed by check and then need to be updated in the config prior to source creation since doing both in a single operation will allow us to make the necessary updates prior to saving the source config to the db.
2. When we write source creation we'll need to make sure we separate that out from the connector config and that we port the frontend logic for complete_oauth response -> discrete fields in source config to the backend so that we don't have to update source specifications
3. As part of the above, in the future frontend will want to be able to run check operations and can't do that unless we have the ability for secret ID -> complete_oauth response -> discrete fields in source config to happen prior to the check. We need to figure out how we'll support that on the frontend.

Reply:
Generally +1 to the message, just adding that the logic for  "given this secret cordinate with all my oauth params, hydrate the configuration with the values in the right spots" should be re-usable since we will need this in create, update, and the various check_connection implementations (there is also a check_connection_for_update)

Reply:
Drive-by:

What does `consider adding a check operation to this process` mean? What is this check process and where does it live?

Reply:
The same check operation that we run prior to creating a source, I guess what pedro calls `check_connection`. Right now as we were thinking about allowing creation of a source, we weren't planning on forcing a check and were initially considering just allowing JSON schema validation to be sufficient, however it seems like that is not in fact sufficient and we should also be running a `check_connection` to help prevent consumers from creating a source with valid spec which can't actually connect.

Pedro also made me aware that we don't run another check unless we get a failed sync which might be confusing/less than ideal user experience for these consumers which is another reason to get the check to happen prior to creating the source and potentially allowing syncs which would fail.

Reply:
makes sense. Did the discussion cover how to integrate check_connection into the server?

Reply:
Loosely from my first post:
> As such, we might benefit from bundling check into source create/update at least externally (for public API consumers).
>    
>    * This has the added benefit of potentially allowing us to handle the internal use case where single use refresh tokens are consumed by check and then need to be updated in the config prior to source creation since doing both in a single operation will allow us to make the necessary updates prior to saving the source config to the db.
(edited for clarity)
I think there's probably more complexity around it than that, but if that's an approach that makes sense then I can at least start thinking in that direction.

Reply:
Ok. Writing the entire response to secret storage seems fine for now. 

Unless, we are going to redo the FE/normal connector flows, I suggest hosting all the checking logic within the public-api server for practical purposes and folding this back into the platform once we know what PBA users like/expect.

Can we leave a javadoc on `writeOAuthSecret` summarising ^? What about renaming from `writeOAuthSecret` to `writeOAuthResponseSecret` to better capture what this is actually doing?

Reply:
@davinchia we do want to do this relatively soon for the regular FE flow - see https://github.com/airbytehq/airbyte-platform-internal/issues/3781

Reply:
I'm definitely +1 on renaming and writing a javadoc to detail what's going on here. Also up for discussing where we can host the logic based on plans for the FE flow and what people generally feel makes the most sense.

Reply:
good to know @pedroslopez . do you also know who is driving that?

Reply:
Perhaps @timroes would have some thoughts on how this can be re-used on the frontend, specially with how we would be storing things in a single blob and would not be "expanding it" frontend-side. From chatting w/ Jon and the team's envisioning the create to just pass in the secret coordinate, it seems like it would be simpler on the FE side since it no longer has to inject the oauth pieces to the right spots in the config, but worth tagging for awareness.